main/acle.md

```diff
@@ -4955,6 +5111,2370 @@ assertions, which may speed up access to aligned data (and will fault
+effect.
+
+### First-faulting and non-faulting loads
+
+SVE provides load instructions in which only the first active element
+can fault, and others in which no elements can fault. A special register
+called the first-fault register (FFR) records which elements were loaded
+successfully. The FFR describes all loads that have executed since the
+last write to the register.
+
+If a first-faulting or non-faulting load does not load an active element
+due to a potential fault, it clears the FFR from that element onwards.
+Those elements of the returned vector then have an unknown value.
+Elements also have an unknown value if the associated FFR element was
+*already* clear before the instruction started.
+
+For example, in:
+
+``` asm
+  SETFFR
+  LDFF1D Z0.D, P0/Z, [X0]
+  LDFF1D Z1.D, P1/Z, [X1]
+  LDFF1D Z2.D, P2/Z, [X2]
+  RDFFR P3.B
+```
+
+the load of `Z0.D` might suppress the load of `[X0, #16]` due to a
+potential fault. It would then clear bit 16 onwards of the FFR and leave
+elements 2 onwards of `Z0.D` in an unknown state. The same elements of
+`Z1.D` and `Z2.D` would then also be unknown, regardless of whether
+the loads based on `X1` or `X2` might fault. At the end of the
+sequence, `P3.B` indicates which elements of `Z0.D`, `Z1.D` and
+`Z2.D` are valid.
+
+In this sequence, the leading inactive elements of `Z0.D` are
+guaranteed to be zero and the first active element is guaranteed to be
+valid (assuming that the first element did not trigger a fault). The
+same guarantees apply to `Z1.D` only if the first active element of
+`P1` is guaranteed to be earlier than the second active element of
+`P0`. In this sense, the contents of `Z0.D`, `Z1.D` and `Z2.D`
+do depend on the order of the instructions, since the guarantees would
+be different if the loads had a different order. However, the point of
+the loads is to try to access more than the first active element, so
+these relationships are not useful in practice.
+
+ACLE therefore divides first-faulting and non-faulting loads (but
+not normal loads) into “FFR groups”. Each group begins and ends with an
+intrinsic that explicitly reads from or writes to the FFR. More precisely,
+ACLE introduces a global “first-fault register token” (FFRT) that identifies
+the current FFR group. This FFRT is a purely conceptual construct and contains
+three pieces of information:
+
+| **Identifier** | **Contents**                                                   |
+| -------------- | -------------------------------------------------------------- |
+| *nwrite*       | the number of explicit writes to the FFR                       |
+| *lastwrite*    | the last value written to the FFR                              |
+| *nread*        | the number of explicit reads from the FFR since the last write |
+
+There are only two possible ways of modifying this FFRT:
+
+1.  Increment *nwrite* by one, set *lastwrite* to a given value, and set
+    *nread* to zero.
+
+2.  Increment *nread* by one.
```

Comment:
```suggestion
*  Increment *nwrite* by one, set *lastwrite* to a given value, and set
    *nread* to zero.

*  Increment *nread* by one.
```

Reply:
I would prefer to keep the enumeration here, since the following text refers to the “first” and “second” operation.